tests/jni_api.rs

```diff
@@ -83,6 +85,27 @@ pub fn get_static_public_field() {
     assert_eq!(min_int_value, i32::min_value());
 }
 
+#[test]
+pub fn get_static_public_field_by_id() {
+    let env = attach_current_thread();
+
+    // One can't pass a JavaType::Primitive(Primitive::Int) to
+    //   `get_static_field_id` unfortunately: #137
+    let field_type = "I";
+    let field_id = env.get_static_field_id(INTEGER_CLASS, "MIN_VALUE", field_type)
+        .unwrap();
+
+    let field_type = JavaType::from_str(field_type).unwrap();
+    let min_int_value = unsafe {
+        env.get_static_field_unsafe(INTEGER_CLASS, field_id, field_type)
+    }
+        .unwrap()
```

Comment:
Personally I don't like formatting here: let's expand `unsafe` to cover all four lines.

Reply:
Now that I think of it — let's add it in the branch where this method is no longer `unsafe` :-)